sw/airborne/modules/ctrl/optical_flow_landing.c

```diff
@@ -225,213 +283,394 @@ void vertical_ctrl_module_run(bool in_flight)
 {
   int i;
   float lp_height; // low-pass height
-  float div_factor; // factor that maps divergence in pixels as received from vision to /frame
+  float div_factor; // factor that maps divergence in pixels as received from vision to 1 / frame
 
   // ensure dt >= 0
   if (dt < 0) { dt = 0.0f; }
 
   // get delta time, dt, to scale the divergence measurements correctly when using "simulated" vision:
   struct timespec spec;
-  clock_gettime(CLOCK_REALTIME, &spec);
-  long new_time = spec.tv_nsec / 1.0E6;
+  clock_gettime(CLOCK_MONOTONIC, &spec);
+  long new_time = spec.tv_sec * 1E3 + spec.tv_nsec / 1.0E6;
   long delta_t = new_time - previous_time;
   dt += ((float)delta_t) / 1000.0f;
   if (dt > 10.0f) {
     dt = 0.0f;
     return;
   }
   previous_time = new_time;
+  long module_active_time = new_time - module_enter_time;
+  float module_active_time_sec = (float) module_active_time / 1000.0f;
+  dt_history[ind_hist % of_landing_ctrl.window_size] = dt;
+  ind_hist++;
 
   if (!in_flight) {
 
     // When not flying and in mode module:
-    // Reset integrators
-    reset_all_vars();
+    // Reset integrators, landing phases, etc.
+    // reset_all_vars(); // commented out to allow us to study the observation variables in-hand, i.e., without flying
+  }
+
+  /***********
+   * VISION
+   ***********/
+
+  if (of_landing_ctrl.VISION_METHOD == 0) {
 
+    // SIMULATED DIVERGENCE:
+
+    // USE OPTITRACK HEIGHT
+    of_landing_ctrl.agl = (float) gps.lla_pos.alt / 1000.0f;
```

Comment:
Where can I find an example of that? 

Grep gives me lots of different kinds of state_gets... For example, I see: mavlink_msg_sim_state_get_alt(msg);
But where do I get msg from?